hamilton/function_modifiers/metadata.py

```diff
@@ -166,3 +167,49 @@ def decorate_node(self, node_: node.Node) -> node.Node:
         new_tags = node_.tags.copy()
         new_tags.update(self.tag_mapping.get(node_.name, {}))
         return tag(**new_tags).decorate_node(node_)
+
+
+SupportedType = Union[
+    int,
+    float,
+    str,
+    bool,
+    dict,
+    list,
+    object,
+    datetime.datetime,
+    datetime.date,
+]
+
+
+def _convert_type_to_str(type_: Type[SupportedType]) -> str:
+    """Converts a type to a string. This is used for the schema decorator.
+
+    :param type_: Type to convert
+    :return: String representation of the type
+    """
+    if type_ in (int, float, str, bool, dict, list, object):
+        return type_.__name__
+    elif type_ == datetime.datetime:
+        return "datetime"
+    elif type_ == datetime.date:
+        return "date"
+    raise ValueError(f"Type {type_} is not supported")
+
+
+class schema(tag):
+    """Represents a schema for a node. This is implemented using tags, but that might change."""
+
+    INTERNAL_SCHEMA_KEY = "hamilton.internal.schema"
```

Comment:
Probably want to call this out explicitly. 
```suggestion
    INTERNAL_SCHEMA_KEY = "hamilton.internal.output_schema"
```